ui/components/app/permission-cell/permission-cell.js

```diff
@@ -114,12 +122,85 @@ const PermissionCell = ({
             className="permission-cell__status"
             variant={TextVariant.bodySm}
             color={TextColor.textAlternative}
+            display={Display.Flex}
           >
             {!revoked &&
               (dateApproved
                 ? t('approvedOn', [formatDate(dateApproved, 'yyyy-MM-dd')])
                 : t('permissionRequested'))}
             {revoked ? t('permissionRevoked') : ''}
+            {accounts && accounts.length && (
+              <Box as="span" display={Display.Flex}>
+                <Text
+                  as="span"
+                  variant={TextVariant.inherit}
+                  color={TextColor.inherit}
+                  marginLeft={1}
+                  marginRight={1}
+                >
+                  {' '}
+                  {t('for')}{' '}
```

Comment:
I don't think this works for languages that have different sentence structure.

Reply:
Do we generally structure translated strings to account for different sentence structures?

Reply:
I think I get it; `t('Requested now') + t'(for')` could be different from `t('Requested now for')`

Reply:
Added some changes to address the request. Storybook will not display the icons anymore (when react components is passed to the translation function), but I added screenshot from the extension modified for testing this use case.

Reply:
Should be resolved in a few refactoring rounds.